src/main/java/org/spongepowered/api/event/entity/living/trader/VillagerEvent.java

```diff
@@ -0,0 +1,106 @@
+package org.spongepowered.api.event.entity.living.trader;
+
+import org.spongepowered.api.data.type.Profession;
+import org.spongepowered.api.entity.Entity;
+import org.spongepowered.api.entity.living.Hostile;
+import org.spongepowered.api.entity.living.golem.IronGolem;
+import org.spongepowered.api.entity.living.monster.raider.Raider;
+import org.spongepowered.api.entity.living.monster.zombie.ZombieEntity;
+import org.spongepowered.api.entity.living.trader.Villager;
+import org.spongepowered.api.event.Cancellable;
+import org.spongepowered.api.event.Event;
+import org.spongepowered.api.raid.Raid;
+import org.spongepowered.api.util.annotation.eventgen.GenerateFactoryMethod;
+
+/**
+ * An event which involves a {@link Villager}.
+ */
+public interface VillagerEvent extends Event {
+
+	/**
+	 * Gets the {@link Villager} involved with this event.\
+	 *
+	 * @return The villager.
+	 */
+	Villager getVillager();
+
+	/**
+	 * Fired when a {@link Villager}'s profession changes.
+	 *
+	 * <p>This can include both gaining an losing a {@link Profession}.</p>
+	 */
+	@GenerateFactoryMethod
+	interface ProfessionChange extends VillagerEvent, Cancellable {
+
+		/**
```

Comment:
Something we do that other APIs really don't is add getOriginalXXXX() to events

getOriginalHealth()
getOriginalExperience()
etc, you get the idea

We do this so a plugin can know what it was to begin with. I'd like to see that in these events.